src/frame/worker/UserCrypto.ts

```diff
@@ -43,6 +43,34 @@ function generateTransformKeyAndEncryptUserKey(userKeySet: KeyPairSet, derivedKe
     });
 }
 
+/**
+ * Requests augmentation on users current private key. The augmented private key is then encrypted and using the password derived key.
+ * @param {Uint8Array} privateKey
+ * @param {Uint8Array} augmentationFactor
+ */
+export function rotatePrivateKey(
+    passcode: string,
+    encryptedPrivateUserKey: Uint8Array,
+    derivedKeySalt: Uint8Array
+): Future<SDKError, {newEncryptedPrivateUserKey: Uint8Array; augmentationFactor: Uint8Array}> {
+    return loadRecrypt()
+        .flatMap((Recrypt) => {
+            return Future.gather2(decryptUserMasterPrivateKey(passcode, encryptedPrivateUserKey, derivedKeySalt), Recrypt.generatePasswordDerivedKey(passcode))
+                .flatMap(([userPrivateKey, derivedKey]) => {
+                    return Future.gather2(Recrypt.rotateUsersPrivateKey(userPrivateKey), Future.of(derivedKey));
+                })
+                .flatMap(([{newPrivateKey, augmentationFactor}, derivedKey]) => {
+                    return AES.encryptUserKey(newPrivateKey, derivedKey).map((encryptedPrivateKey) => {
+                        return {
+                            newEncryptedPrivateUserKey: encryptedPrivateKey,
+                            augmentationFactor,
+                        };
```

Comment:
Minor, but you can avoid the `return` in this section by doing

```
return AES.encryptUserKey(newPrivateKey, derivedKey).map((encryptedPrivateKey) => ({
    newEncryptedPrivateUserKey: encryptedPrivateKey,
    augmentationFactor,
});
```

Reply:
done